cartographer/io/xray_points_processor.cc

```diff
@@ -83,17 +83,24 @@ XRayPointsProcessor::XRayPointsProcessor(const double voxel_size,
       transform_(transform),
       voxels_(voxel_size, Eigen::Vector3f::Zero()) {}
 
-void XRayPointsProcessor::Process(const PointsBatch& batch) {
-  for (const auto& point : batch.points) {
+void XRayPointsProcessor::Process(std::unique_ptr<PointsBatch> batch) {
+  for (const auto& point : batch->points) {
     const Eigen::Vector3f camera_point = transform_ * point;
     *voxels_.mutable_value(voxels_.GetCellIndex(camera_point)) = true;
   }
-  next_->Process(batch);
+  next_->Process(std::move(batch));
 }
 
-void XRayPointsProcessor::Flush() {
+PointsProcessor::FlushResult XRayPointsProcessor::Flush() {
   WriteImage();
-  return next_->Flush();
+  switch (next_->Flush()) {
+    case FlushResult::kRestartStream:
+      LOG(FATAL) << "Not supported";
```

Comment:
"X-Ray generation must be configured to occur after any stages that require multiple passes."